cmake/extensions.cmake

```diff
@@ -346,6 +346,7 @@ endmacro()
 macro(zephyr_library)
   zephyr_library_get_current_dir_lib_name(lib_name)
   zephyr_library_named(${lib_name})
+  zephyr_library_sources(${ZEPHYR_BASE}/misc/empty_file.c)
```

Comment:
Triggering a build failure when a library has no sources is not a bug, it is a feature.

Removing this feature will cause problems.

For instance, users use the build log to see what is included in their build, if they believe they have
disabled the UART driver, but still see an uart lib being linked in then this will cause confusion.

We should not make things more difficult for our users in the name of making our own testing easier.

Reply:
The vanilla behavior of cmake when a cmakelists.txt is empty is to continue and not fail. This is what users expect. This issue is a by-product of how zephyr macros operate, not an intended feature. 

> We should not make things more difficult for our users in the name of making our own testing easier.

we are making it more difficult by changing to cmake to work differently than the default behavior.

Reply:
I don't understand what you mean by a cmakelists.txt being empty.

The default behaviour of CMake is to error-out at generation-time when a library does not have any sources.

> CMake Error at CMakeLists.txt:21 (add_library):
>   No SOURCES given to target: foo
>

Reply:
If you could help me by describing more precisely the problem being solved, preferably with an example, then maybe we can figure something out.

Right now I am just seeing a change, a vague problem description, and a problem that is created by this change.

Reply:
forgot about the library requiring at least one source, let me see what other options are available.

Reply:
remove specific commit for now.